File: serve.js
#!/usr/bin/env node
/**
 * Simple HTTP server to serve the Webflow static website locally
 */
const http = require('http');
const fs = require('fs');
const path = require('path');
const url = require('url');

const PORT = 8000;

// MIME types for different file extensions
const mimeTypes = {
  '.html': 'text/html',
  '.css': 'text/css',
  '.js': 'text/javascript',
  '.json': 'application/json',
  '.png': 'image/png',
  '.jpg': 'image/jpeg',
  '.jpeg': 'image/jpeg',
  '.gif': 'image/gif',
  '.svg': 'image/svg+xml',
  '.avif': 'image/avif',
  '.ico': 'image/x-icon',
  '.webp': 'image/webp'
};

function getMimeType(filePath) {
  const ext = path.extname(filePath).toLowerCase();
  return mimeTypes[ext] || 'text/plain';
}

function serveFile(res, filePath) {
  fs.readFile(filePath, (err, data) => {
    if (err) {
      res.writeHead(404, { 'Content-Type': 'text/html' });
      res.write('<h1>404 - Archivo no encontrado</h1>');
      res.end();
      return;
    }

    const mimeType = getMimeType(filePath);
    res.writeHead(200, { 
      'Content-Type': mimeType,
      'Access-Control-Allow-Origin': '*'
    });
    res.write(data);
    res.end();
  });
}

const server = http.createServer((req, res) => {
  let parsedUrl = url.parse(req.url);
  let pathname = parsedUrl.pathname;
  
  // Default to index.html for root path
  if (pathname === '/') {
    pathname = '/index.html';
  }
  
  const filePath = path.join(__dirname, pathname);
  
  // Check if file exists
  fs.access(filePath, fs.constants.F_OK, (err) => {
    if (err) {
      // Try to serve index.html for client-side routing
      serveFile(res, path.join(__dirname, 'index.html'));
    } else {
      serveFile(res, filePath);
    }
  });
});

server.listen(PORT, () => {
  console.log('🚀 Servidor iniciado en el puerto', PORT);
  console.log('📱 Abre tu navegador y ve a: http://localhost:' + PORT);
  console.log('🌐 También puedes usar: http://127.0.0.1:' + PORT);
  console.log('⏹️  Presiona Ctrl+C para detener el servidor');
  console.log();
});

server.on('error', (err) => {
  if (err.code === 'EADDRINUSE') {
    console.log('❌ Error: El puerto', PORT, 'ya está en uso');
    console.log('💡 Intenta cerrar otros servidores o usa otro puerto');
  } else {
    console.log('❌ Error al iniciar el servidor:', err);
  }
});

process.on('SIGINT', () => {
  console.log('\n🛑 Servidor detenido');
  process.exit(0);
}); 

File: serve.py
#!/usr/bin/env python3
"""
Simple HTTP server to serve the Webflow static website locally
"""
import http.server
import socketserver
import os
import sys
from pathlib import Path

def main():
    PORT = 8000
    
    # Change to the directory containing the HTML files
    os.chdir(Path(__file__).parent)
    
    # Custom handler to serve files with proper MIME types
    class CustomHTTPRequestHandler(http.server.SimpleHTTPRequestHandler):
        def end_headers(self):
            self.send_header('Access-Control-Allow-Origin', '*')
            self.send_header('Access-Control-Allow-Methods', 'GET, POST, OPTIONS')
            self.send_header('Access-Control-Allow-Headers', 'Content-Type')
            # Prevent caching to ensure changes are always visible
            self.send_header('Cache-Control', 'no-cache, no-store, must-revalidate')
            self.send_header('Pragma', 'no-cache')
            self.send_header('Expires', '0')
            super().end_headers()
        
        def guess_type(self, path):
            # Handle common web file types
            if path.endswith('.css'):
                return 'text/css'
            elif path.endswith('.js'):
                return 'text/javascript'
            elif path.endswith('.svg'):
                return 'image/svg+xml'
            elif path.endswith('.avif'):
                return 'image/avif'
            return super().guess_type(path)
    
    try:
        with socketserver.TCPServer(("", PORT), CustomHTTPRequestHandler) as httpd:
            print(f"🚀 Servidor iniciado en el puerto {PORT}")
            print(f"📱 Abre tu navegador y ve a: http://localhost:{PORT}")
            print(f"🌐 También puedes usar: http://127.0.0.1:{PORT}")
            print(f"⏹️  Presiona Ctrl+C para detener el servidor")
            print()
            
            httpd.serve_forever()
    except KeyboardInterrupt:
        print("\n🛑 Servidor detenido")
        sys.exit(0)
    except OSError as e:
        if e.errno == 48:  # Address already in use
            print(f"❌ Error: El puerto {PORT} ya está en uso")
            print(f"💡 Intenta cerrar otros servidores o usa otro puerto")
        else:
            print(f"❌ Error al iniciar el servidor: {e}")
        sys.exit(1)

if __name__ == "__main__":
    main() 